### DISCUSSION TITLE
16.1.x or 17.1.x Upgrade?

### DISCUSSION THREAD

**Discussion by dphu-cg** [keycloak/keycloak#11335] [Q&A]
We are currently on KC v15.1.x  using Wildfly.   We are now planning to upgrade to the later/latest KC version.  Question: Should we plan to upgrade to 16.1.x first before going to 17.1.x?   Or it might be worth effort planning the upgrade directly from 15 to 17?  Wonder if anyone can share their experience upgrading 15 -> 17.

**Comment by tacerus** [comment]
I'm interested in the same. Since Keycloak 18 was released last week, I would particularly be interested if we can upgrade from 15.1 to 18.0 directly or if we should take one or more intermediary steps.

@dphu-cg Have you possibly found an answer elsewhere in the meantime, or did any testing?
